### PR TITLE
Fixed overlap between button text and icon

### LINKED ISSUE #8899: Trusted book provider "audiobook" link in carousel on mobile has text overlap with external icon
### Problem

Hard to explain but see screenshot of "audiobook" button

#### Evidence / Screenshot
Screen Shot 2024-03-12 at 17 44 14

#### Relevant URL(s)
https://openlibrary.org/works/OL37473900W/Short_Story_Collection_102

### Reproducing the bug

* Expected behavior: No overlap
* Actual behavior: Overlap

### Context

- Browser (Chrome, Safari, Firefox, etc): FF
- OS (Windows, Mac, etc): Android
- Logged in (Y/N): Y
- Environment (prod, dev, local): prod

#### Proposal & constraints

If you want to work on this please confirm that you will read the feedback and progress in the closed PR https://github.com/internetarchive/openlibrary/pull/9066 and incorporate it into the PR you open.

#### Stakeholders

@cdrini

### PR DESCRIPTION
**Pull request by Realmbird**
Closes #8899

Fixes overlap between button text and icon

### Technical

- Issue: Audio book read buttons had button text & icon "external link" overlap on smaller screen sizes
- Cause: Responsive booksPerBreakpoints were not accurate with when less pages should be shown on the page

- Fix: Changing responsive booksPerBreakpoints was not css like issue had stated

### Testing

1. docker compose up
// For running site
2. docker compose exec -e PYTHONPATH=. web bash -c "./scripts/copydocs.py --search 'author_key:OL9411725A' --search-limit 100"
// Adding data
3. http://localhost:8080/works/OL36891831W/
4. Start changing screen width

### Screenshot

[Screencast from 04-10-2024 01:54:07 PM.webm](https://github.com/internetarchive/openlibrary/assets/56459277/823fe90a-010f-47ff-9da5-183c9692d1b4)
Will be screencast to demonstrate that text never overlaps with icon

### Stakeholders

@cdrini @RayBB

### CODE REVIEW

**Comment by RayBB** [comment]
@Realmbird it is on [testing](https://testing.openlibrary.org/works/OL36891831W/Short_Mystery_Suspense_Collection_013) now and it is working as expected. I think it fixes the initial concern. However, I want to run it by the design team in the call today. 👍

**Comment by Realmbird** [comment]
Thanks

**Comment by RayBB** [comment]
The design call didn't have time for me to bring this up.

Maybe @cdrini can comment directly.

**Comment by RayBB** [comment]
@Realmbird we have to wait for @cdrini to comment on this one. He's off Thursday and Friday so it will probably have to wait until next week.

Here are my thoughts though:
- Your PR does fix the issue
- However, I'm not sure if those numbers you changed in the JS were set for a specific reason so I'm hesitant to change them.
- In my opinion, I think it is a better long-term fix to move the image out of the background and have it be right next to the text. But that's just my opinion and changes the design of the button. I assume the original goal was that the external icon stays to the far right while the text stays centered. I'm not sure how attached we are to that design.

**Comment by RayBB** [comment]
@Realmbird 
I asked in the design call today. 
The feedback I got from @mekarpeles was that maybe we should see if we can make the text/icon scale smaller with the size of the button. We are still not sure if that is feasible and probably want to get feedback from @cdrini before spending too much time on that path.

**Comment by RayBB** [comment]
@Realmbird feel free to play around with moving the icon out but ultimately we should wait for Drini's input here. We can try to get his input on the Tuesday call.

**Comment by Realmbird** [comment]
Ok in the mean time gonna work on other issue

**Comment by cdrini** [comment]
Howdy folks! I think what we can try is; apologies it will be a little involved!

1) Decrease the font size of all `cta-btn` from 16px to 14px. Make only the primary button, the one inside the `#read-options` element, be 16px.
2) Make all `.cta-btn` `white-space: nowrap; overflow: hidden; text-overflow: ellipsis;` and make `.cta-btn--external` have a padding-right of 25px ; this should make sure there's enough space for the external icon, and crop the text with an ellipsis if there isn't
3) Make all buttons have `padding: 8px 0` instead of the current `padding: 8px` ; the button label already has padding, so we're just doubling up on it.
4) Make all buttons have `text-align: center`; this is getting lost somewhere for carousels.

That should do it, and give us something like this

image

Here's the css I experimented with ; although yours will likely look different / be in a different spot

image

**Comment by Realmbird** [comment]
Ok will make changes to css

**Comment by Realmbird** [comment]
@cdrini Added your changes

**Comment by Realmbird** [comment]
[Screencast from 05-22-2024 11:22:57 AM.webm](https://github.com/internetarchive/openlibrary/assets/56459277/d9ffd12f-0544-4b23-8d56-e3f0abce4df0)
After changes

File: openlibrary/plugins/openlibrary/js/carousel/Carousel.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class Carousel {
             speed: 300,
             slidesToShow: this.config.booksPerBreakpoint[0],
             slidesToScroll: this.config.booksPerBreakpoint[0],
-            responsive: [1200, 1024, 600, 480, 360]
+            responsive: [1200, 1024, 740, 580, 360]
```
Comment (RayBB) [inline openlibrary/plugins/openlibrary/js/carousel/Carousel.js:69]:
```suggestion
            responsive: [1200, 1024, 600, 480, 360]
```
Lets restore these (I think that's what Drini wanted) then I'll put it on testing so we can see how things look!

**Review by RayBB — CHANGES_REQUESTED** [review]

**Comment by Realmbird** [comment]
@RayBB @cdrini  Tried rays changes caused overlap bug to reappear

**Comment by RayBB** [comment]
Closing this to offer up someone else to work on this one.

**Pull request closed without merging**